conda-store-server/conda_store_server/api.py

```diff
@@ -82,6 +83,176 @@ def update_namespace(
     return namespace
 
 
+# v2 API
+def update_namespace_metadata(
+    db,
+    name: str,
+    metadata_: Dict[str, Any] = None,
+):
+    namespace = get_namespace(db, name)
+    if namespace is None:
+        raise ValueError(f"Namespace='{name}' not found")
+
+    if metadata_ is not None:
+        namespace.metadata_ = metadata_
+
+    return namespace
+
+
+# v2 API
+def get_namespace_roles(
+    db,
+    name: str,
+):
+    """Which namespaces can access namespace 'name'?"""
+    namespace = get_namespace(db, name)
+    if namespace is None:
+        raise ValueError(f"Namespace='{name}' not found")
+
+    nrm = aliased(orm.NamespaceRoleMappingV2)
+    this = aliased(orm.Namespace)
+    other = aliased(orm.Namespace)
+    q = (
+        db.query(nrm.id, this.name, other.name, nrm.role)
+        .filter(nrm.namespace_id == namespace.id)
+        .filter(nrm.namespace_id == this.id)
+        .filter(nrm.other_namespace_id == other.id)
+        .all()
+    )
+    return [schema.NamespaceRoleMappingV2.from_list(x) for x in q]
+
+
+# v2 API
+def get_other_namespace_roles(
+    db,
+    name: str,
+):
+    """To which namespaces does namespace 'name' have access?"""
+    namespace = get_namespace(db, name)
+    if namespace is None:
+        raise ValueError(f"Namespace='{name}' not found")
+
+    nrm = aliased(orm.NamespaceRoleMappingV2)
+    this = aliased(orm.Namespace)
+    other = aliased(orm.Namespace)
+    q = (
+        db.query(nrm.id, this.name, other.name, nrm.role)
+        .filter(nrm.other_namespace_id == namespace.id)
+        .filter(nrm.namespace_id == this.id)
+        .filter(nrm.other_namespace_id == other.id)
+        .all()
+    )
+    return [schema.NamespaceRoleMappingV2.from_list(x) for x in q]
+
+
+# v2 API
+def delete_namespace_roles(
+    db,
+    name: str,
+):
+    namespace = get_namespace(db, name)
+    if namespace is None:
+        raise ValueError(f"Namespace='{name}' not found")
+
+    nrm = orm.NamespaceRoleMappingV2
+    db.query(nrm).filter(nrm.namespace_id == namespace.id).delete()
+
+
+# v2 API
+def get_namespace_role(
+    db,
+    name: str,
+    other: str,
+):
+    namespace = get_namespace(db, name)
```

Comment:
This block appears a lot of times. It may be worth extracting it into something like `get_existing_namespace_by_name` or something like that which wraps `get_namespace` and raises a valueerror if the name doesn't exist.

This is a stylistic suggestion and not a blocker if you choose not to implement it.

Reply:
I agree in general, but not going to change this. I had similar code in the original PR that I abstracted out into a common function. I was later told in a meeting that copy-pasting is preferred. Because it allows reviewers to focus on part of the code without referencing anything else and helps with refactoring since each part is independent from one another.

Reply:
I disagree with that reasoning, but ultimately it is stylistic and not a big enough deal to worry too much about.